Gems/AzQtComponentsForPython/CMakeLists.txt

```diff
@@ -0,0 +1,224 @@
+set(shiboken_options --generator-set=shiboken --enable-parent-ctor-heuristic
+    --enable-pyside-extensions --enable-return-value-heuristic --use-isnull-as-nb_nonzero
+    --avoid-protected-hack --language-level=c++17
+    --license-file=${LICENSE_FILE}
+    ${INCLUDES}
+    -I${CMAKE_SOURCE_DIR}
+    -I${CMAKE_CURRENT_LIST_DIR}/..
+    -I"${LY_ROOT_FOLDER}/code/Framework/AzCore"
+    -I"${LY_ROOT_FOLDER}/code/Framework/AzQtComponents"
+    -I"${PROJECT_BINARY_DIR}/../AzQtComponents_autogen/include"
+    -I"${LY_ROOT_FOLDER}/code/Framework/AzCore/Platform/${PAL_PLATFORM_NAME}"
+    -I"${PYSIDE2_INCLUDE_DIR}"
+    -I"${SHIBOKEN_INCLUDE_DIR}"
+    -I"${PYSIDE2_INCLUDE_DIR}/QtWidgets"
+    -I"${PYSIDE2_INCLUDE_DIR}/QtCore"
+    -I"${PYSIDE2_INCLUDE_DIR}/QtGui"
+    -I"${PYSIDE2_TYPESYSTEM_PATH}"   
+    -I"${QT_MODULE_DIR}/include"
+    -I"${QT_MODULE_DIR}/include/QtWidgets"
+    -I"${QT_MODULE_DIR}/include/QtCore"
+    -I"${QT_MODULE_DIR}/include/QtGui"
+    -T"${PYSIDE2_PATH}/typesystems"
+    -T"${PYSIDE2_PATH}"
+    -T"${SHIBOKEN_INCLUDE_DIR}"
+    -T"${PYSIDE2_TYPESYSTEM_PATH}"
+    -T"${PYSIDE2_SHARE_PATH}"
+    --output-directory=${CMAKE_CURRENT_BINARY_DIR}
+)
+
+set(generated_sources_dependencies ${wrapped_header} ${typesystem_file})
+
+# Dummy library to run the custom shiboken command.
+ly_add_target(
+    NAME ${BINDINGS_LIBRARY}.bindings STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        dummy_files.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            ${CMAKE_CURRENT_LIST_DIR}/..
+            ${PYSIDE2_INCLUDE_DIR}
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzQtComponents
+            Gem::EditorPythonBindings.Static
+)
+
+# Add the custom shiboken command to the dummy library.
+add_custom_command(TARGET ${BINDINGS_LIBRARY}.bindings
+                    POST_BUILD
+                    COMMAND ${SHIBOKEN_GENERATOR_EXE_PATH} ${shiboken_options} ${wrapped_header} ${typesystem_file}
+                    DEPENDS ${generated_sources_dependencies} 
+                        Gem::AzQtComponentsForPython.WrappedFiles
+                    COMMENT "Running generator for ${typesystem_file}.")
+    
+# Create dummy wrapper files so the second project has something to work with while being created.
+include(wrapped_files.cmake)
+foreach(WRAPPER_FILE ${WRAPPED_FILES})
+    if(NOT EXISTS "${WRAPPER_FILE}")
+        file(WRITE ${WRAPPER_FILE} "")
+    endif()
+endforeach()
+    
+# Set up the project for building the code generated by shiboken.
+ly_add_target(
+    NAME ${BINDINGS_LIBRARY} SHARED
+    NAMESPACE Gem
+    FILES_CMAKE
+        wrapped_files.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            ${PYSIDE2_INCLUDE_DIR}
+            ${PYTHON_INCLUDE_DIR}
+            ${SHIBOKEN_INCLUDE_DIR}
+            "${PYSIDE2_INCLUDE_DIR}\\QtCore"
+            "${PYSIDE2_INCLUDE_DIR}\\QtGui"
+            "${PYSIDE2_INCLUDE_DIR}\\QtWidgets"
+            "${PYSIDE2_INCLUDE_DIR}\\QtNetwork"
```

Comment:
CMake uses forward slashes for pathing.

Also include directories are propagated from the PUBLIC dependencies of a dependent target.
So adding a dependency on 3rdParty::pyside2 and 3rdParty::python would make those include directories available for the bindings library target.

Reply:
Slashes fixed, it looks like PySide isn't handling setting up the include paths correctly at the moment so I'll leave it like that unless it gets fixed.

Reply:
Well this is going to be problem on Linux